### ISSUE TITLE
Use dndbeyond.com to host character sheets?

### ISSUE THREAD

**Issue by crimsonknave** [our-stories/west-marches#8]
I've used dndbeyond.com for multiple campaigns and it works well. It's the officially branded site. I have almost every non adventure book on there and can share them all with anyone in the campaign (without anyone else having to pay). 

https://www.dndbeyond.com/characters/32922888 is an example character sheet on there. I don't think I've encountered a situation where the character sheet didn't work or something was missing (you can create custom backgrounds and feats and so on).

According to https://www.dndbeyond.com/subscriptions you can only have 12 players in a campaign (I'm fairly sure that isn't a limit on characters, just players to prevent one person sharing all the content with thousands of people). If that's a limit we hit we could have two campaigns in theory.

**Comment by zmoazeni** [comment]
I most definitely want this conversation to continue and _not_ be shut down on any short term decisions I make here.

I will come back and give some more coherent thoughts. But here's a screenshot from some of my sentiments from Slack:

![image](https://user-images.githubusercontent.com/314/89199590-c4a48a00-d57c-11ea-878b-5eb4048250e7.png)

**Comment by chrisbloom7** [comment]
What would be the reason for asking everyone to use this tool, and it is worth forcing folks to learn a new tool and/or have to maintain two copies of information? There is undoubtedly a base set of information that is useful for the GM to have available at a glance, so what subset of character info is worth keeping updated in some common system or format with the least amount of friction to keep updated, assuming the rest of that info could be a click away in a format that's easy for players to upload or export when they level up or there's some other significant change.

**Comment by zmoazeni** [comment]
> There is undoubtedly a base set of information that is useful for the GM to have available at a glance, so what subset of character info is worth keeping updated in some common system or format with the least amount of friction to keep updated...

I do think it's worthwhile to have other Players be able to browse through the Characters while as they progress too. I think that's just great transparency that encourages others to join in. Getting started is less of a pain because they can see most of the process.

As a GM, I _do_ need access to all the characters because I will be building encounters off of the average (or median) levels. But I will also want to throw in some stuff that would allow each party member to shine. It really stinks to have `Animal Handling` and _never_ be able to use it right? Or specific utility spells where a Cleric is like _"FINALLY! THE TIME HAS COME!!"

**Comment by crimsonknave** [comment]
> What would be the reason for asking everyone to use this tool, and it is worth forcing folks to learn a new tool and/or have to maintain two copies of information? 

I thought it would be a nice central place for all characters to be stored. If folks don't want to use it that's fine, just throwing it out as an option (and noting that I'm already paying for it so it shouldn't cost anything for folks).

I'd be happy with any solution that allows for players and the GM to see the character sheets of other players. I'd ideally like to see the full sheet for everyone as that would help me recall who ended up with specific loot, or check to see if someone has a specific spell and then suggest that it could be helpful in the current situation.

**Comment by zmoazeni** [comment]
Yeah all good on the suggestion @crimsonknave!

I'm even considering putting together a small ruby script or go program that given a text or PDF export would automatically update the Wiki page for your character in this Repo.

I think that's why I'm leaning towards the GitHub Wiki being the central place right now. It feels the easiest to maintain. Now I would actually prefer it committed into the repo itself, but adding/editing Wiki pages is slightly more friendly than adding a new file to a directory in the repo through the UI. Not all players will be devs with git knowledge, and I don't want it to be a requirement.

**Comment by crimsonknave** [comment]
We'll want to avoid storing lots of versions PDFs in git... That'll get big fast, but the wiki or a pages site would work (just store the PDFs or images elsewhere).

**Comment by zmoazeni** [comment]
A plan came to me. Rather than using GitHub Issues (and markdown templates) or GitHub Wikis we can just share files outside of GitHub.

* l can setup a shared publicly readable OneDrive.
* All players can use whatever tooling they want to use and send a link to a text file (or if it's plain, post it in the issue), RTF file, a PDF, a Google doc, Google Spreadsheet. It doesn't matter. As long as I can download it. **Also most PDFs should be under the GitHub file upload limit, so the simplest route would be to attach a PDF to the issue!**
* Players can submit updates through issues with a new attachment or link whenever they want to update their character sheet. _As a GM I ask that it is updated regularly so I can build appropriate adventures and encounters._
* Here's the clever part: I can setup a GitHub Action so that this transfer can all be automated. Perhaps I'll roll together a simple Go script that will allow players to submit these issues via CLI.

Then it doesn't matter what character system you use. We don't need to edit the character sheets in GitHub, we just need them easily accessible by GMs and Players alike.

I'm feeling pretty good to on this approach. And in the meantime character issues can simplified by pointing people in the direction of external resources and then dropping links into the character issue for approval. And I'll deal with the storage manually.

It would be 

Anyone strongly opposed or have any ideas on how to improve ☝️ ?

**Comment by crimsonknave** [comment]
So, my only tweak to this would be that I think it would be good to have a .md file in the repo that links to each character's sheet. That link could be to a PDF stored in OneDrive or any publicly visible version of the character (dndbeyond link, link to another character sheet site or a link to a text file). If the goal is to allow folks to create a character wherever/however they feel most comfortable I think that would fit it well. It'll also provide for a place to store a small bio/blurb about the character and then a link to their sheet so there's an easy to find, central place for the party.

**Comment by zmoazeni** [comment]
> I think it would be good to have a .md file in the repo that links to each character's sheet.
> 
> It'll also provide for a place to store a small bio/blurb about the character and then a link to their sheet so there's an easy to find, central place for the party.

I like that a lot! And the session issues can link directly to those bios. Great idea.

**Comment by djigneo** [comment]
I think there's value to committing to a specific platform (even if that means "we're not using a virtual tabletop"). Speaking as a potential GM.

1. Simpler for players to transition between GMs (who may be using different platforms).
2. Consistent toolset for players and GMs for running games.

If we're not going to use a Virtual tabletop, then having a central repository of (editable) character sheets and linked from github seems fine (as long as they're in one place). Even in this case, I think it makes sense to have an consistent, editable character sheet.

**Comment by zmoazeni** [comment]
Yeah the more I use dndbeyond, I have to admit that I admire it more.

The thing I hesitate on is that I know some folks prefer things like https://www.wolflair.com and if something better (possibly Open Source with perhaps an API!?) comes out I don't want to have to change things all up around character management. I'm hoping that PDF versions will be that manual stop-gap.

My opinion right now is to sit with the pain to see how bad it gets, and if it's just too much we can lock down to 1, 2, n services. I'm okay with being wrong here.

**Comment by djigneo** [comment]
I'm ok with that, I think it should just be a consistent character sheet format.

**Comment by zmoazeni** [comment]
Ugh. I just came across the campaign book sharing, the [discord bot](https://avrae.io/), and the [twitch extensions](https://features.dndbeyond.com/twitch-extension/).

I'm starting to think maybe we _should_ just make it consistent and ask everyone to use dndbeyond.

I know I have been the strong voice of dissent there. But now I'm thinking "why the hell am I wading upstream?" **This comment is to make it clear I do NOT have a strong opinion on this anymore.**

Does anyone else think we _shouldn't_ use dndbeyond.com?

**Comment by chrisbloom7** [comment]
One thing that is already frustrating to me after using it for a day is that some things like spells, backgrounds, and subclasses are limited to the OGL content unless you pay to unlock the digital source books, which I wouldn't have a problem with if they weren't $30 each for books I already spent $40 or $50 on. If I could manually enter these things it wouldn't be so bad, but I don't see a way to do that with spells or subclasses - you have to choose from the list they give you.

**Comment by chrisbloom7** [comment]
That said, I appreciate the formatting and how easy the character generation is, but for some builds I may end up only using it for base stats and then tracking the rest elsewhere.

**Comment by crimsonknave** [comment]
@chrisbloom7 I posted an invite link in discord, join the campaign and you'll have access to all my books for any character in the campaign

**Comment by zmoazeni** [comment]
I also purchased the bundles. So for games that I GM, I can invite folks into ad-hoc campaigns now too. That will be a good thing to formalize in some fashion if we do make this a requirement.

**Comment by chrisbloom7** [comment]
Thanks @crimsonknave. I was able to join, import my first PC, and update the spell list to include the one's that weren't available previously. It looks like I can only import (link) one of my characters with the campaign, but I see there's an option to `CREATE UNASSIGNED CHARACTER` in the campaign. Creating a new character outside the campaign still limits the selection to the OGL content. Is the assumption here that we'll create all of our characters in the campaign? Is this going to scale well? Not trying to shoot down the idea because of some inconvenience, just want to make sure we're all on the same page and those of us who aren't ready to drop a few benjamins on the source books are willing to deal with some friction. (And bless those of you who did and are willing to share through the campaign)

**Comment by crimsonknave** [comment]
I'm not :100: on the mechanics of it, but I do know that I can share with 5 campaigns (I need one for another campaign) and that you can have 12 players in each campaign. I was able to add a second character to the campaign by going to the join link again (which is pretty un-intuitive/clunky). The pricing page says I can share with 12 players, I was able to add a bunch of random characters I had lying about and confirmed that we can have more than 12 characters (which makes sense for NPCs and so on).

@chrisbloom7 Let me know if you aren't able to create more characters in the campaign by visiting the join link multiple times.

**Comment by chrisbloom7** [comment]
@crimsonknave ok, I'll give that a try. I created an unassigned test character inside the campaign. I wonder if anyone can claim it and edit it, or if it's still linked to the creating account?

**Comment by crimsonknave** [comment]
I had the option of claiming it when I got the link.
